qlib/portfolio/optimizer/base.py

```diff
@@ -0,0 +1,13 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+import abc
+
+
+class BaseOptimizer(abc.ABC):
+    """ Construct portfolio with a optimization related method """
+
+    @abc.abstractmethod
+    def __call__(self, *args, **kwargs) -> object:
```

Comment:
Can we define the interface (input & output) for all optimizers here?

Reply:
For example:
```python
class BaseOptimizer(abc.ABC):
    """ Construct portfolio with a optimization related method """

    @abc.abstractmethod
    def __call__(
        self,
        S: Optional[Union[np.ndarray, pd.DataFrame, Tuple]] = None,
        u: Optional[Union[np.ndarray, pd.Series]] = None,
        w0: Optional[Union[np.ndarray, pd.Series]] = None,
        w_bench : Optional[Union[np.ndarray, pd.Series]] = None,
        inds_onehot : Optional[Union[np.ndarray, pd.DataFrame]] = None,
    ) -> Union[np.ndarray, pd.Series]:
```

Reply:
After discussion, we decided to save those changes to the next pull request.